src/flatland/ordered/map.clj

```diff
@@ -86,7 +86,8 @@
     (.equiv this other))
   (hashCode [this]
     (reduce (fn [acc ^MapEntry e]
-              (let [k (.key e), v (.val e)]
+              (let [k (if-some [k (.key e)] k 0)
```

Comment:
Am I right in assuming that was a typo and you're asking me to implement mapHash (not hashMap) as per here:

https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/APersistentMap.java#L103-L112

If so, coming right up :)